lib/reredos.rb

```diff
@@ -33,5 +29,32 @@ def valid_username?(str)
       return false if str.length > USERNAME_MAX_LENGTH
       /\A[0-9a-zA-Z\.\+\-\_]+\z/ === str # 既定の文字のみで構成されている
     end
+
+    # ラベルはアルファベットで始まり、アルファベットか数字かハイフンが続き、アルファベットか数字で終わる
+    def valid_label?(str)
+      return false if str.empty?
+      return false if str.length > DOMAIN_LABEL_MAX_LENGTH
+      return false unless letter?(str[0])
+      return false unless alphanumeric?(str[-1])
+      return true if str[1...-1].empty?
+
+      str[1...-1].chars.all? { |c| hyphen_or_alphanumeric?(c) }
+    end
+
+    def letter?(char)
+      ('a'..'z').include?(char) || ('A'..'Z').include?(char)
```

Comment:
ここを
```suggestion
      ('A'..'z').include?(char)
```
としないのは可読性のためでしょうか？

Reply:
```
irb(main):001:0> ('A'..'z').to_a
=> ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "[", "\\", "]", "^", "_", "`", "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
```

なので `'A'..'z'` では許可してはならない文字が含まれてしまうからです。

Reply:
なるほど、おっしゃるとおりでした 確認不足が多くすみません

マージします 👍